File: applications/sinks/mqtt-log-sink/src/main/java/com/vmware/data/solutions/rabbitmq/streaming/MqttConfig.java
package com.vmware.data.solutions.rabbitmq.streaming;

import lombok.extern.slf4j.Slf4j;
import org.eclipse.paho.client.mqttv3.*;
import org.eclipse.paho.client.mqttv3.persist.MemoryPersistence;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.nio.charset.StandardCharsets;

/**
 * MqttConfig
 *
 * @author Gregory Green
 */
@Configuration
@Slf4j
public class MqttConfig
{

    @Value("${spring.application.name:mqtt-log-sink}")
    private String clientId = "mqtt-log-sink";

    @Value("${mqtt.connectionUrl:tcp://localhost:1883}")
    private String connectionUrl = "tcp://localhost:1883";

    @Value("${mqtt.userName:guest}")
    private String userName = "mqtt";

    @Value("${mqtt.userPassword:guest}")
    private String userPassword = "mqtt";

    @Value("${mqtt.topic.filter:test}")
    private String topicFilter;

    @Bean
    IMqttClient mqttClient() throws MqttException
    {
        var mqttClient = new MqttClient(connectionUrl, clientId, new MemoryPersistence());

        var options = new MqttConnectOptions();
        options.setAutomaticReconnect(true);
        options.setCleanSession(true);
        options.setConnectionTimeout(10);
        options.setUserName(userName);
        options.setPassword(userPassword.toCharArray());
        mqttClient.connect(options);

        IMqttMessageListener listener = (topic, mqttMessage) -> {
            String body = new String(mqttMessage.getPayload(), StandardCharsets.US_ASCII);
            log.info("TOPIC: {} BODY: {}",topic,body);
        };
        mqttClient.subscribe(topicFilter,listener);

        return mqttClient;
    }

}

File: applications/generator-mqtt-source/src/main/kotlin/com/vmware/tanzu/data/services/mqtt/generatormqttsource/MqttConfig.kt
package com.vmware.tanzu.data.services.mqtt.generatormqttsource

import nyla.solutions.core.util.Config
import org.eclipse.paho.client.mqttv3.IMqttClient
import org.eclipse.paho.client.mqttv3.MqttClient
import org.eclipse.paho.client.mqttv3.MqttConnectOptions
import org.eclipse.paho.client.mqttv3.persist.MemoryPersistence
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableScheduling

/**
 * @author Gregory Green
 */
@Configuration
@EnableScheduling
class MqttConfig {
    @Value("\${spring.application.name:generator-mqtt-source}")
    private var publisherId: String = "generator-mqtt-source";

    @Value("\${mqtt.connectionUrl:tcp://localhost:1883}")
    private var connectionUrl: String = "tcp://localhost:1883";

    @Value("\${mqtt.userName:mqtt}")
    private val userName: String = "mqtt";

    @Value("\${mqtt.userPassword:mqtt}")
    private val userPassword: String = "mqtt";

    @Bean
    fun mqttClientPublisher() : IMqttClient
    {
        val publisher: IMqttClient = MqttClient(connectionUrl, publisherId, MemoryPersistence())

        val options = MqttConnectOptions()
        options.isAutomaticReconnect = true
        options.isCleanSession = true
        options.connectionTimeout = 10
        options.userName = userName
        options.password = userPassword.toCharArray()
        publisher.connect(options)

        return publisher

    }

    @Bean("template")
    fun template() : String{
        return Config.getProperty("GENERATOR_TEMPLATE")
    }

}